libp2p/builders.nim

```diff
@@ -16,7 +16,10 @@ runnableExamples:
    # etc
    .build()
 
-{.push raises: [Defect].}
+when (NimMajor, NimMinor) < (1, 4):
```

Comment:
There is https://github.com/markspanbroek/upraises precisely to work around this, I think it's thin enough that it can be used without much issues?

Reply:
`upraises` was also mentioned and rejected by @arnetheduck in the [nimbus PR](https://github.com/status-im/nimbus-eth2/pull/3888), because:

> ... they introduce a lot of complexity for what ultimately is a cosmetic problem that will go away soon